jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-client/src/main/java/org/jbpm/workbench/cm/client/newcase/NewCaseInstanceViewImpl.java

```diff
@@ -151,13 +153,24 @@ public void clearRoles() {
     }
 
     @Override
-    public void setCaseDefinitions(final List<String> definitions) {
+    public void setCaseDefinitions(final List<CaseDefinitionSummary> definitions) {
         clearCaseDefinitions();
         caseTemplatesList.setValue("");
-        for (final String definition : definitions) {
-            caseTemplatesList.addOption(definition);
-        }
+        definitions.forEach(d -> {
+            final Optional<CaseDefinitionSummary> cds = definitions.stream().filter(dn -> dn.getName().equals(d.getName()) && dn.getUniqueId().equals(d.getUniqueId()) == false).findFirst();
```

Comment:
This condition seems rather confusing:
`dn.getName().equals(d.getName()) && dn.getUniqueId().equals(d.getUniqueId()) == false`

It took me a while to get the point :) (It's distinguishing case definition names which determine CD uniquely from the names that have more different CDs associated, right?)

Apart from readability I see other problem with this way of checking "which name identify the case uniquely" and that is quadratic complexity. For each case definition all other case definitions are checked. E.g. if you have 100 case definitions it will check 100x100 pairs of definitions etc.

The same functionality can be achieved by single linear pass through the list of definitions and aggregating them into the map by name. Like:
```
Map<String,List<Definitions> nameToDefinitions = definitions.stream().collect(Collectors.groupingBy(CaseDefinitionSummary::getName));

nameToDefinitions.entrySet().forEach(entry -> {
  if (entry.value().size() > 1) {
     // add disambiguated case def.
  } else {
     //add uniquely identified case def.
  }
});
```
@cristianonicolai WDYT?